test/unit/contentChunk.js

```diff
@@ -159,14 +160,72 @@ describe('contentChunk: ', () => {
         }
       })
       const payloadLength = payload.length
-  
+
       const docId = await extractStringPayloadIntoChunks(payload)
-  
+
       db.collection('fs.files').findOne({_id: docId}, (err, result) => {
         should.ok(result)
         should.deepEqual(result._id, docId)
         should.deepEqual(result.length, payloadLength)
       })
     })
   })
+
+  describe('retrievePayload()', () => {
+    it('should return an error when the file id is null', (done) => {
+        const fileId = null
+
+        retrievePayload(fileId, (err, body) => {
+          err.message.should.eql(`Payload retrieval failed: Payload id: ${fileId} is invalid`)
+          done()
+        })
+    })
+
+    it('should return the body', (done) => {
+      const bucket = new mongodb.GridFSBucket(db)
+      const stream = bucket.openUploadStream()
+      const fileString = `JohnWick,BeowulfJohnWick,BeowulfJohnWick,BeowulfJohnWick,Beowulf
+                          JohnWick,BeowulfJohnWick,BeowulfJohnWick,BeowulfJohnWick,Beowulf
+                          JohnWick,BeowulfJohnWick,BeowulfJohnWick,BeowulfJohnWick,Beowulf
+                          JohnWick,BeowulfJohnWick,BeowulfJohnWick,BeowulfJohnWick,Beowulf
+                        `
+
+      stream.on('finish', (doc) => {
+        if(doc) {
+          const fileId = doc._id
+
+          retrievePayload(fileId, (err, body) => {
+            body.should.eql(fileString)
+            done()
+          })
+        }
+      })
+
+      stream.end(fileString)
+    })
+
+    it('should return an error and null when file does not exist', (done) => {
```

Comment:
The description of this test states that `null` is also returned which can be removed as we not testing for `null` and only care about the err object